query-engine/driver-adapters/js/adapter-neon/src/conversion.ts

```diff
@@ -116,6 +118,79 @@ export function fieldToColumnType(fieldTypeId: number): ColumnType {
   }
 }
 
+/****************************/
+/* Time-related data-types  */
+/****************************/
+
+
+function normalize_date(date: string): string {
+  return date
+}
+
+function normalize_timestamp(time: string): string {
+  return time
+}
+
+function normalize_timestampz(time: string): string {
+  return time.split("+")[0]
+}
+
+function normalize_array(element_normalizer: (string) => string): (string) => string[] {
+  return function(str: string): string[] {
+    return parseArray(str, element_normalizer)
+  }
```

Comment:
Although functionally equivalent in this case, it would be a little bit more idiomatic to use an arrow function for a closure like this.

I think it's also fine to drop the type annotations from the closure since there's already an explicit return type on the line above:

```suggestion
  return (str) => parseArray(str, element_normalizer)
```

Reply:
Very nice usage of currying for the factory of array parsers though 💯